server/chat-plugins/rock-paper-scissors.ts

```diff
@@ -0,0 +1,399 @@
+		clearTimeout(this.roundTimer);
+		this.addControls(`The game is paused.`);
+		this.add(`The game is paused.`);
+	}
+	unpause(user: User) {
+		const player = this.getPlayer(user);
+		if (this.roundTimer) throw new Chat.ErrorMessage(`The game is not paused.`);
+		this.room.add(Utils.html`|html|${player.name} unpaused the game.`).update();
+		this.startNextRound();
+	}
+	startNextRound() {
+		this.currentRound++;
+		if (this.currentRound > 0) {
+			this.addField(
+				`Round ${this.currentRound} of ${this.rounds} has begun! ` +
+				`Players, you have ${Chat.toDurationString(TIMEOUT)} to make your moves!`
+			);
+		}
+		this.room.add(`|html|<h2>Round ${this.currentRound}</h2>`).update();
+		this.sendOptions();
+		this.roundTimer = setTimeout(() => {
+			this.runMatch();
+		}, TIMEOUT);
+	}
+	destroy() {
+		if (this.roundTimer) clearTimeout(this.roundTimer);
+		this.room.pokeExpireTimer();
+		this.addControls(`The game has ended.`);
+		this.ended = true;
+		this.room.add(`The game has been forcefully ended.`); // for the benefit of those in the room
+		this.room.log.log = [];
+		for (const id in this.playerTable) {
+			this.playerTable[id].unlinkUser();
+		}
+		this.playerTable = {};
+	}
+	choose(user: User, option: string) {
+		const player = this.getPlayer(user);
+		if (player.currentChoice) throw new Chat.ErrorMessage("You have already made your choice!");
+		player.currentChoice = option;
+		this.add(Utils.html`${user.name} has made their choice.`);
+		this.resetOptions(user);
+		if (this.players.filter(item => item.currentChoice).length > 1) {
+			clearTimeout(this.roundTimer!);
+			return this.runMatch();
+		}
+		return true;
+	}
+	leaveGame(user: User) {
+		const player = this.getPlayer(user);
+		player.sendRoom(`You left the game.`);
+		delete this.playerTable[user.id];
+		this.end();
+	}
+	addPlayer(user: User) {
+		if (this.playerTable[user.id]) throw new Chat.ErrorMessage(`You are already a player in this game.`);
+		this.playerTable[user.id] = new RPSPlayer(user, this);
+		this.room.auth.set(user.id, Users.PLAYER_SYMBOL);
+		user.sendTo(this.room, `You have successfully joined the Rock, Paper, Scissors game.`);
+		return this.playerTable[user.id];
+	}
+}
+
+function getGame(context: CommandContext) {
+	const room = context.requireRoom();
+	const game = room.getGame(RPSGame);
+	if (!game) throw new Chat.ErrorMessage(`There is no Rock, Paper, Scissors game going on in this room.`);
+	if (game.gameid !== 'rockpaperscissors') {
+		throw new Chat.ErrorMessage(`A game of ${game.title} is already going on.`);
+	}
+	return game;
+}
+
+function findExisting(user1: string, user2: string) {
+	return Rooms.get(`rps-${user1}-${user2}`) || Rooms.get(`rps-${user2}-${user1}`);
+}
+
+export const commands: ChatCommands = {
+	rps: 'rockpaperscissors',
+	rockpaperscissors: {
+		challenge: 'create',
+		create(target, room, user) {
+			room = this.requireRoom();
```

Comment:
Why does this require a room? I thought you were supposed to challenge in PM?

Reply:
Mistake.

Reply:
It doesn't require a pm or a challenge, though, it'll just pm the challenge.